crates/project_panel/src/project_panel.rs

```diff
@@ -1533,6 +1544,7 @@ impl ProjectPanel {
         sanitized_entries: BTreeSet<SelectedEntry>,
         cx: &mut Context<Self>,
     ) -> Option<SelectedEntry> {
+        let hide_gitignore = ProjectPanelSettings::get_global(cx).hide_gitignore;
```

Comment:
nit: Can be move down just before where it is being used. So, that if we early return below we don't need to store this.